home/views.py

```diff
@@ -0,0 +1,21 @@
+from django.shortcuts import render
+from django.urls import reverse
+from django.views.generic import TemplateView
+
+class TestView(TemplateView):
+    template_name = "home/test.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data()
+        context["id"] = "lorem ipsum"
+        return context
+        
+class ServiceWorkerView(TemplateView):
```

Comment:
I would prefer it if the service worker was just a normal static JS file that fetches the list of URLs to pre-cache from a server endpoint.

Reply:
Yep, I agree and this was one of the first approaches I tried. Unfortunately, I ran into some issues. 

The service worker must essentially be defined in a JavaScript file that must be served at the root of the application. If it is under /static its URL will be something like /static/sw.js. It means that the scope of the ServiceWorker (ie the pages it is activated on), must be under /static. This means that it won't be enabled for our whole site (which is what we want, limiting it to /static makes it unusable on standard pages so it would be quite useless). So it must be served from something like http://wwww.example.com/sw.js.

To achieve this, I created a simple TemplateView that will serve a file named sw.js. This appears to be the best way to achieve what we set out to do